akka-actor-tests/src/test/scala/akka/util/ConfigValidationSpec.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.util
+
+import akka.actor.DefaultSupervisorStrategy
+import akka.testkit.AkkaSpec
+import com.typesafe.config.ConfigFactory
+
+class ConfigValidationSpec extends AkkaSpec {
+  import ConfigFactory.parseString
+  import ConfigValidation._
+
+  "ConfigValidation" must {
+    val empty = ConfigFactory.empty
+    val config = system.settings.config
+    val emptyConfigValue = parseString("""akka.empty = "" """)
+
+    "fail fast in validation if an expected plugin is missing" in {
+      intercept[ConfigValidationError](config.plugin("akka.missing.fqcn"))
+      intercept[ConfigValidationError](config.plugin(null))
+      intercept[ConfigValidationError](config.plugin(""))
+      intercept[ConfigValidationError](empty.plugin("akka.missing.fqcn"))
+    }
+
+    "return valid plugin if config is valid" in {
+      val plugin = config.plugin("akka.actor.guardian-supervisor-strategy")
+      plugin shouldBe classOf[DefaultSupervisorStrategy]
+    }
+
+    "fail fast with 'valueOrThrow' if invalid" in {
+      intercept[ConfigValidationError](config.valueOrThrow[String]("akka.missing")(stringConfigVal))
+      intercept[ConfigValidationError](config.valueOrThrow[Boolean]("akka.missing")(booleanConfigVal))
+      intercept[ConfigValidationError](empty.valueOrThrow[String]("akka.empty")(stringConfigVal))
+      intercept[ConfigValidationError](empty.valueOrThrow[Boolean]("akka.empty")(booleanConfigVal))
+    }
+
+    "convert value with 'valueOrThrow' if valid" in {
+      config.valueOrThrow[String]("akka.loglevel")(stringConfigVal) shouldBe "WARNING"
```

Comment:
@patriknw this is an example of no boilerplate in if/else/matches where the code author has written/will write `throw IllegalArgumentException` but would not have to. It could be a one-liner and return the `Class`. I just did `Class.forName` as one quick way, expecting this idea to be shot down. But I like it.